### PR TITLE
Cache nonce, retry on nonce errors

### PR DESCRIPTION
**Pull request by vgrichina**

### CODE REVIEW

**Comment by vgrichina** [comment]
WARNING: no specific tests yet. Normal tests are technically covering the code, but best to test specific situations.

File: src/account.ts [hunk 1]
```diff
@@ -144,28 +112,55 @@ export class Account {
...
-        );
+            const nonce = ++accessKey.nonce;
+            [txHash, signedTx] = await signTransaction(
+                receiverId, nonce, actions, base_decode(status.sync_info.latest_block_hash), this.connection.signer, this.accountId, this.connection.networkId
+            );
+
+            try {
+                const result = await exponentialBackoff(TX_STATUS_RETRY_WAIT, TX_STATUS_RETRY_NUMBER, TX_STATUS_RETRY_WAIT_BACKOFF, async () => {
+                    try {
+                        return await this.connection.provider.sendTransaction(signedTx);
```
Comment (bowenwang1996) [inline src/account.ts:134]: If I understand this logic correctly, we should not retry that frequently, especially at the beginning of the backoff.

Comment (vgrichina) [reply src/account.ts:134]: @bowenwang1996 why? We are only doing new `broadcast_tx_commit` call after previous one returned timeout (my understanding is that it takes 10 seconds).

Comment (bowenwang1996) [reply src/account.ts:134]: ah I got confused.

**Pull request merged**